crates/sui-framework/packages/move-stdlib/tests/option_tests.move

```diff
@@ -169,4 +169,52 @@ module std::option_tests {
         let v: vector<u64> = option::none().to_vec();
         assert!(v.is_empty());
     }
+
+    // === Macros ===
+
+    #[test]
+    fun do_destroy() {
+        let mut counter = 0;
+        option::some(5).destroy!(|x| counter = x);
+        option::some(10).do!(|x| counter = counter + x);
+
+        assert!(counter == 15);
```

Comment:
I snuck in `std::unit_test::assert_eq` a while back if you want to give it a shot. This is fine though